collectd/files/plugin/openstack_nova.py

```diff
@@ -52,18 +52,30 @@ def collect(self):
                                  'service': service,
                                  'state': state})
 
-        for service in aggregated_workers:
-            totalw = sum(aggregated_workers[service].values())
+        for service in set(aggregated_workers.keys()).union(
+                ('compute', 'scheduler', 'conductor', 'cert', 'consoleauth')):
+
+            if service in aggregated_workers:
+                totalw = sum(aggregated_workers[service].values())
 
             for state in self.states:
-                prct = (100.0 * aggregated_workers[service][state]) / totalw
-                self.dispatch_value('nova_services_percent', '',
-                                    prct,
-                                    {'state': state, 'service': service})
-
-                self.dispatch_value('nova_services', '',
-                                    aggregated_workers[service][state],
-                                    {'state': state, 'service': service})
+                if service in aggregated_workers:
```

Comment:
since aggregated_workers is a `defaultdict(Counter)`, you don't need a special case:

```
>>> from collections import Counter
>>> from collections import defaultdict
>>> f = defaultdict(Counter)
>>> f['compute']['up'] = 3
>>> f['compute']['up']
3
>>> f['compute']['down']                                                                                                                                                                                                               
0
>>> f['compute2']['down']                                                                                                                                                                                                              
0
>>> f['compute4']['down']                                                                                                                                                                                                              
0
>>> f['compute4']['up']
0
```